### PR TITLE
enhance: add search params in search request in restful

### PR DESCRIPTION
**Pull request by smellthemoon**
https://github.com/milvus-io/milvus/issues/36321

### CODE REVIEW

File: internal/distributed/proxy/httpserver/request_v2.go [hunk 1]
```diff
@@ -141,18 +141,29 @@ type CollectionDataReq struct {
 
 func (req *CollectionDataReq) GetDbName() string { return req.DbName }
 
+type searchParams struct {
+	// // not use metricType any more, just for compatibility
+	MetricType      string                 `json:"metricType"`
+	Params          map[string]interface{} `json:"params"`
+	IgnoreGrowing   bool                   `json:"ignore_growing"`
+	PageRetainOrder bool                   `json:"page_retain_order"`
```
Comment (PwzXxm) [inline internal/distributed/proxy/httpserver/request_v2.go:149]: this should be in SearchParams->`Params`, not outside.

**Comment by smellthemoon** [comment]
related with #36321

File: internal/distributed/proxy/httpserver/handler_v2.go [hunk 2]
```diff
@@ -905,61 +905,50 @@ func generatePlaceholderGroup(ctx context.Context, body string, collSchema *sche
...
-		PartitionNames:        httpReq.PartitionNames,
-		UseDefaultConsistency: true,
+		DbName:         dbName,
+		CollectionName: httpReq.CollectionName,
+		Dsl:            httpReq.Filter,
+		DslType:        commonpb.DslType_BoolExprV1,
+		OutputFields:   httpReq.OutputFields,
+		PartitionNames: httpReq.PartitionNames,
+	}
+	if httpReq.ConsistencyLevel != "" {
```
Comment (PwzXxm) [inline internal/distributed/proxy/httpserver/handler_v2.go:928]: Is it possible to extract this and reuse at below line [997](https://github.com/milvus-io/milvus/pull/36304/files#diff-b81c4eb20e49c64dc253ff54abb0a4d56aace2d6e63fd78b9be98f312c9a678bR997)

**Comment by smellthemoon** [comment]
rerun go-sdk

**Review by congqixia — APPROVED** [review]
/approve

**Comment by smellthemoon** [comment]
@zhuwenxing could you help to update the related tests in restful?

File: internal/distributed/proxy/httpserver/request_v2.go [hunk 3]
```diff
@@ -141,18 +141,28 @@ type CollectionDataReq struct {
 
 func (req *CollectionDataReq) GetDbName() string { return req.DbName }
 
+type searchParams struct {
+	// not use metricType any more, just for compatibility
+	MetricType    string                 `json:"metricType"`
+	Params        map[string]interface{} `json:"params"`
+	IgnoreGrowing bool                   `json:"ignore_growing"`
```
Comment (zhuwenxing) [inline internal/distributed/proxy/httpserver/request_v2.go:148]: Do we need to maintain consistency in naming conventions around here?
metric_type --> metricType
so ignore_growing --> ignoreGrowing?

**Pull request merged**